scripts/biomedical/NCBI_ClinVar/scripts/format_clinvar_and_position.py

```diff
@@ -0,0 +1,1209 @@
+
+    writeEntry2('dbVarID', dict_info, 'DBVARID', w, True)
+
+    write_clin_sig_category(dict_info, 'ONC',
+                            'clinicalSignificanceCategory_onc', w)
+
+    write_review_status(dict_info, 'ONCREVSTAT',
+                        'clinVarOncogenicityReviewStatus', w)
+
+    write_clin_sig_category(dict_info, 'SCI',
+                            'clinicalSignificanceCategory_sic', w)
+
+    write_review_status(dict_info, 'SCIREVSTAT', 'clinVarSomaticReviewStatus',
+                        w)
+    #print("line", '\n', line, '\n', "CURRENT_ROW", CURRENT_ROW)
+
+    return w, dict_info, dcid, rsID, CURRENT_ROW
+
+
+def writeNode_POS(line, count, ASSEMBLY_HG19, start_time):
+    w = WriteToCsv()
+    parse_clinvar_pos_row(line, w, None)
+    # print("POSline", line, '\n', CURRENT_ROW)
+    #w.write('\n')
+    w.write_to_file(['clinvar_pos_only'])
+    w.close()
+    count += 1
+    if count % 100000 == 0:  # start writing to new output file
+        time_running = time.time() - start_time
+        logging.info(f"{count} lines processed: {time_running:.2f} second")
+    return (count)
+
+
+def parse_clinvar_pos_row(line, w, curr_row=None):
+    global CURRENT_ROW
+    if curr_row:
+        CURRENT_ROW = copy.deepcopy(curr_row)
+
+    l = line[7].split(';')
+    dict_info = {}
+    dcid = 'bio/VariationID_' + line[2]
+    for item in l:
+        entry = item.split('=')
+        dict_info[entry[0]] = entry[1]
+    if 'RS' in dict_info:
+        dcid = 'bio/rs' + dict_info['RS']
+        rsID = dict_info['RS']
+        w.write('dcid', f"dcid:{dcid}")
+        w.write('name', f'{rsID}')
+        w.write('rsID', rsID)
+    elif line[2].isdigit():
+        w.write('dcid', f"dcid:{dcid}")
+        w.write('name', f'{line[2]}')
+    else:
+        logging.info(line)
+
+    w.write('dcid_pos', f'bio/{ASSEMBLY_HG19}_chr{line[0]}_{line[1]}')
+    w.write('name_pos', f'"{ASSEMBLY_HG19} chr{line[0]} {line[1]}"')
+    w.write('chrom', f'chr{line[0]}')
+    w.write('position', line[1])
+    return CURRENT_ROW
+
+
+def process_clinvar_files():
+    global CURRENT_ROW, OUTPUT_CLINVAR_FILE_NAME, OUTPUT_CLINVAR_CONFLICTING_FILE_NAME, OUTPUT_CLINVAR_OBS_FILE_NAME
+    global CLINVAR_FILE_WRITER, CLINVAR_CONFLICTING_FILE_WRITER, CLINVAR_OBS_FILE_WRITER
+    OUTPUT_CLINVAR_FILE_NAME = os.path.join(MODULE_DIR,
+                                            OUTPUT_CLINVAR_FILE_NAME)
+    OUTPUT_CLINVAR_CONFLICTING_FILE_NAME = os.path.join(
+        MODULE_DIR, OUTPUT_CLINVAR_CONFLICTING_FILE_NAME)
+    OUTPUT_CLINVAR_OBS_FILE_NAME = os.path.join(MODULE_DIR,
+                                                OUTPUT_CLINVAR_OBS_FILE_NAME)
+    # set start time
+    start_time = time.time()
+    # open all output files and save in global scope
+    clinvar_file, conflicting_file, obs_file = None, None, None
+    try:
+        clinvar_file = open(OUTPUT_CLINVAR_FILE_NAME, 'w')
+        CLINVAR_FILE_WRITER = csv.DictWriter(clinvar_file,
```

Comment:
dcid has inappropriate dcid prefix in the clinvaroutput file - see note above on dcids. geneID property value also has inappropriate dcid prefix - the Gene node should be initialized in the tMCF with the dcid for tave the property geneID this node type pointing to the output of this column. Then for GeneticVariant node will point to the initialized instance of the Gene node for that row. The same should occur for SequenceOntologyTerms nodes as well.